### PR TITLE
Log errors od

### PR DESCRIPTION
**Pull request by reshke**
Screenshot from 2020-07-07 17-43-47

### CODE REVIEW

File: sources/counter.h [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+#include <malloc.h>
+#include <kiwi.h>
+#include <zconf.h>
+#include "macro.h"
+
+typedef struct od_counter od_counter_t;
+typedef size_t od_hash_table_key_t;
+
+#define OD_DEFAULT_HASH_TABLE_SIZE 15
```
Comment (x4m) [inline sources/counter.h:19]: I think this must be near Hashtable , right?

File: sources/counter.h [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+od_hash_item(od_counter_t *t, od_hash_item_t item)
+{
+	/*
+	 * How to get hash of number in range 0 to MAXINT ?
+	 * there are many ways, we choose simplest one
+	 */
+	return item % t->size;
+}
+
+/* these are function to perform dynamic hash table resize in case of overfit
```
Comment (x4m) [inline sources/counter.h:96]: Is it counter or is it hashtable?

File: sources/status.h [hunk 3]
```diff
@@ -9,24 +9,24 @@
...
-	OD_EOOM,
-	OD_EATTACH,
-	OD_EATTACH_TOO_MANY_CONNECTIONS,
-	OD_ESERVER_CONNECT,
-	OD_ESERVER_READ,
-	OD_ESERVER_WRITE,
-	OD_ECLIENT_READ,
-	OD_ECLIENT_WRITE
-} od_status_t;
+	OD_UNDEF                        = 0,
```
Comment (x4m) [inline sources/status.h:12]: Are those numbers significant? Just curious...

File: sources/counter.h (hunk 1 above)
Comment (reshke) [reply sources/counter.h]: yeah, right

File: sources/frontend.c [hunk 4]
```diff
@@ -208,9 +210,12 @@ od_frontend_attach(od_client_t *client,
 				         client,
 				         NULL,
 				         "server pool wait timed out, closing");
-				return OD_EATTACH_TOO_MANY_CONNECTIONS;
```
Comment (x4m) [inline sources/frontend.c:211]: What has actually changed here?

File: sources/console.c [hunk 5]
```diff
@@ -191,7 +305,93 @@ od_console_show_stats(od_client_t *client, machine_msg_t *stream)
...
+		case OD_PARSER_EOF:
+		default:
+			return OD_GENERAL_NOTOK_RESPONSE_CODE;
+	}
+	od_keyword_t *keyword;
+	keyword = od_keyword_match(od_console_keywords, &token);
+	if (keyword == NULL)
+		return -1;
+	switch (keyword->id) {
+		case OD_LFRONTEND:
```
Comment (x4m) [inline sources/console.c:388]: Do we really really want two different views for FE and BE errors? Maby just show them in different lines of the same view?

Comment (reshke) [reply sources/console.c]: merged into one

File: sources/frontend.c (hunk 4 above)
Comment (reshke) [reply sources/frontend.c:211]: reverted

**Pull request merged**